clientcontroller/evm_consumer.go

```diff
@@ -128,14 +273,55 @@ func (ec *EVMConsumerController) QueryActivatedHeight() (uint64, error) {
 }
 
 func (ec *EVMConsumerController) QueryLatestBlockHeight() (uint64, error) {
-	/* TODO: implement
-	get the latest L2 block number from a RPC call
-	*/
 
-	return uint64(0), nil
+	type Block struct {
+		Number string
+	}
+
+	var block Block
+	err := ec.evmClient.Call(&block, "eth_getBlockByNumber", "latest", true)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get latest block:%s", err)
+	}
+
+	num, err := strconv.ParseUint(strings.TrimPrefix(block.Number, "0x"), 16, 64)
+	if err != nil {
+		fmt.Println("Error:%s", err)
+	}
+
+	return num, nil
+
 }
 
 func (ec *EVMConsumerController) Close() error {
 	ec.evmClient.Close()
 	return nil
 }
+
+func Transform(number *big.Int) string {
+
+	hexStr := fmt.Sprintf("%x", number)
+	if len(hexStr) >= 2 && hexStr[:2] != "0x" {
+		hexStr = "0x" + hexStr
+	}
+	return hexStr
+
+}
+func (ec *EVMConsumerController) GetLatestFinalizedNumber() (*big.Int, error) {
```

Comment:
missing a new line after the function closure

Reply:
Add a line